### PR TITLE
feat: [CDS-99919]: Schema changes for support for delegate virtual service in Traffic routing

### PR DESCRIPTION
**Pull request by namantalaycha10**
Edit Schema Changes for support for delegate virtual service in Traffic routing
in this, we are adding a boolean field in IstioProviderSpec 
and when it is marked as true we will not populate the host

### CODE REVIEW

File: v0/pipeline/steps/cd/k8s-istio-provider-spec.yaml [hunk 1]
```diff
@@ -23,4 +23,10 @@ allOf:
           - type: string
             pattern: "^<\\+input>((\\.)((executionInput\\(\\))|(allowedValues|default|regex)\\(.+?\\)))*$"
             minLength: 1
+      delegateService:
+        oneOf:
+          - type: boolean
+          - type: string
+            pattern: "^<\\+input>((\\.)((executionInput\\(\\))|(allowedValues|default|regex)\\(.+?\\)))*$"
```
Comment (abosii-harness) [inline v0/pipeline/steps/cd/k8s-istio-provider-spec.yaml:30]: This pattern supports only runtime input, we also want to support expressions

Comment (namantalaycha10) [reply v0/pipeline/steps/cd/k8s-istio-provider-spec.yaml]: Done

**Review by abosii-harness — DISMISSED** [review]

File: v0/pipeline.json [hunk 2]
```diff
@@ -45656,6 +45656,15 @@
                     "pattern" : "^<\\+input>((\\.)((executionInput\\(\\))|(allowedValues|default|regex)\\(.+?\\)))*$",
                     "minLength" : 1
                   } ]
+                },
```
Comment (vlica-harness) [inline v0/pipeline.json:45659]: @namantalaycha10 I think we should add this to v1 as well

Comment (namantalaycha10) [reply v0/pipeline.json:45659]: Thanks for Mentioning , added the changes

**Review by vlica-harness — APPROVED** [review]

**Review by vikyathharekal — APPROVED** [review]

**Review by ivan-83 — APPROVED** [review]

**Pull request merged**